### PR TITLE
Optimize map gestures - transmit function

### PR DESCRIPTION
**Pull request by Sniperhid**
Better worst case complexity.

### CODE REVIEW

File: addons/map_gestures/functions/fnc_transmit.sqf [hunk 1]
```diff
@@ -28,18 +28,18 @@ if (!GVAR(EnableTransmit) || !visibleMap) exitWith {
...
-        [EVENT_PLAYER_HAS_NO_OWNER_ID, [name _x]] call EFUNC(common,serverEvent);
-    } else {
-        _playerOwnerID = ACE_player getVariable QGVAR(owner_id);
-        if (!isNil "_playerOwnerID" && _ownerID != _playerOwnerID) then {
-            _unitUID = getPlayerUID ACE_Player;
-            _drawPosVariableName = if (!isNil "_unitUID" && _unitUID != "") then {format [QGVAR(%1_DrawPos), _unitUID]} else {nil};
-            if (!isNil "_drawPosVariableName") then {
+_unitUID = getPlayerUID ACE_Player;
+_playerOwnerID = ACE_player getVariable QGVAR(owner_id);
+if (!isNil "_unitUID" && _unitUID != "" && !isNil "_playerOwnerID") then {
```
Comment (commy2) [inline addons/map_gestures/functions/fnc_transmit.sqf]: `!isNil "_unitUID" && _unitUID != ""` intended to use lazy eval?

File: addons/map_gestures/functions/fnc_transmit.sqf [hunk 2]
```diff
@@ -28,18 +28,18 @@ if (!GVAR(EnableTransmit) || !visibleMap) exitWith {
...
+    _drawPosVariableName = format [QGVAR(%1_DrawPos), _unitUID];
+    {
+        _ownerID = _x getVariable QGVAR(owner_id);
+        if (isNil "_ownerID") then {
+            [EVENT_PLAYER_HAS_NO_OWNER_ID, [name _x]] call EFUNC(common,serverEvent);
+        } else {
+            if (_ownerID != _playerOwnerID) then {
                 _ownerID publicVariableClient _drawPosVariableName;
```
Comment (commy2) [inline addons/map_gestures/functions/fnc_transmit.sqf]: Why do we even have this in the first place? All transmitting should be done with the ACE event system.

Comment (MikeMatrix) [inline addons/map_gestures/functions/fnc_transmit.sqf]: It was the first port. Will be working on the event based system and some other optimizations for this soon'ish.

**Comment by PabstMirror** [comment]
I guess we put this on back burner, sorry Sniperhid.
But I just tested and it didn't seem to be working (could be wrong, `getPlayerUID` makes it tough to test).

```
_playerOwnerID = ACE_player getVariable QGVAR(owner_id);
```
If that is nil, then we skip the for loop, but the only way we set that var is from an event inside the loop.

I think just adding this should fix:
```
if (isNil "_playerOwnerID") exitWith {
    [EVENT_PLAYER_HAS_NO_OWNER_ID, [name ACE_Player]] call EFUNC(common,serverEvent);
};
```

**Comment by commy2** [comment]
Yeah, any scripting around `getPlayerUID` is bad. It's similar to addressing machines by their `profileName` or `name player`, or objects by their `vehicleVarName`.
The correct way is to use owner Id's and object references.

**Comment by Glowbal** [comment]
Closing, merge conflicts, inactivity

**Pull request closed without merging**